src/software/ai/hl/stp/play/offense/offense_play_test.py

```diff
@@ -67,18 +69,21 @@ def test_offense_play(simulated_test_runner):
     )
 
     # Always Validation
-    # TODO- #2779 Validation
-    always_validation_sequence_set = [[]]
+    inv_always_validation_sequence_set = [
+        [BallAlwaysStaysInRegion(regions=[field.fieldBoundary()])]
+    ]
+
+    ag_always_validation_sequence_set = [[FriendlyAlwaysHasBallPossession()]]
```

Comment:
I was thinking whether this will run into problems when a shot is taken on goal.